### ISSUE TITLE
Using Skaffold and LocalStack to Develop Cloud Applications Locally

### ISSUE THREAD

**Issue by gergelyke** [gergelyke/gergelyke.github.io#17]

**Comment by tarasowski** [comment]
Hi Gergely,

thank you for your intro to localstack. What do you think, what is currently the best workflow for developing serverless apps on AWS?

Thanks,
Dimitri

**Comment by gergelyke** [comment]
Hi Dimitri,

thanks! For serverless, I'd use the [Serverless Framework](https://serverless.com/)

I hope that helps!

**Comment by efi-mk** [comment]
Hello @gergelyke ,
In your original article you mentioned "_Should each developer create a development environment for themselves in the cloud?_" and answered that there is a drawback to this method, can you elaborate a bit more ? Is it complicated ? does it cost a lot of money ? What are the main obstacles for a developing firm to this that method ?

Thank you

**Comment by gergelyke** [comment]
Hi @efi-mk! 

There can be a list of drawbacks for setting up development environments for everybody in the cloud:

* it can become costly,
* it will be challenging to keep it updated (afaik, there is no off-the-shelf solution for this, you have to build your own)

I hope this answers your question! Please let me know if it did not, and I am happy to provide more details.

**Comment by efi-mk** [comment]
Thank you @gergelyke, I completely agree, but I have couple of problems with using only localstack for testing:
* It does not pose all the limitations that a native environment has
* It's not updated with the latest api changes
* What should I do in case I use a service that is not listed there like image recognition or NLP capbilities

For me it looks like testing in the actual native serverless environment is a must before actual deployment. Makes sense ?

Do you think that _off-the-shelf solution_ would make tools like LocalStack obsolete ?

**Comment by gergelyke** [comment]
Hi @efi-mk - yeah, that makes sense, I agree with your concerns. As i see, you have to distinguish between local development versus automated testing.

For local development, I'd use LocalStack with Skaffold - for automated testing, I'd provision everything on AWS, and run the functional tests on that infrastructure.

Does that make sense?

**Comment by efi-mk** [comment]
Makes sense. @gergelyke is there a way to contact you directly? I have couple of questions regarding your testing methodology.

Thank you for your advice

**Comment by gergelyke** [comment]
My DMs are open on twitter: https://twitter.com/nthgergo

**Comment by demisx** [comment]
Hi @gergelyke. Thank you for a very helpful post. I was able to get the localstack running with skaffold v0.29, but I don't see any logs in the output, even though I set the `DEBUG env var (see below) and I've confirmed that this var is set in the container too. Any idea how to get to the localstack logs? Thank you.

```yaml
     env:
            - name: SERVICES
              value: 's3:30572,lambda:30574'
            - name: PORT_WEB_UI
              value: '32000'
            - name: DEBUG
              value: '1'
```

**Comment by demisx** [comment]
I think I got it going. Looks like in order to for logs to start showing up in skaffold output, the Docker image must be built by skaffold itself, even if the Dockerfile contains just one line:

```Dockerfile
FROM localstack/localstack
```

**Comment by gergelyke** [comment]
hi @demisx - glad to hear that you liked the post! Since the writing of the article, I've barely used localstack, so I am not sure if I can help, I am sorry

**Comment by demisx** [comment]
@gergelyke Yeah, I've got the log part going, but it's not a smooth ride switching endpoint from AWS to localstack. And lack of documentation doesn't help either. I see errors popping up left and right and my AWS usage is very light. Did you find a better alternative to the `localstack` by any chance?

**Comment by lxm7** [comment]
Can Scaffold or the more recent LocalStack get round issue of supporting multi regions these days? If not can anyone suggest a good alternative to these in order to cater for multi regional setups? Thanks
